main.go

```diff
@@ -235,10 +228,28 @@ func sendToWriter(ingestionType string, msgBody io.Reader, uuid string, URLSlice
 	if err != nil {
 		log.Errorf("Cannot read error body: [%v]", err)
 	}
-
 	return fmt.Errorf("reqURL=[%s] status=[%d] uuid=[%s] error=[%v] body=[%s]", reqURL, resp.StatusCode, uuid, reqErr, string(errorMessage))
 }
 
+func resolveWriterAndCreateRequest(ingestionType string, msgBody io.Reader, uuid string, URLSlice []string) (*http.Request, string, error) {
+	var writerURL string
+	for _, URL := range URLSlice {
+		if strings.Contains(URL, ingestionType) {
+			writerURL = URL
+		}
+	}
+	if writerURL == "" {
+		return nil, "", fmt.Errorf("No configured writer for concept: %v", ingestionType)
+	}
+	reqURL := writerURL + "/" + ingestionType + "/" + uuid
```

Comment:
The reason I didnt add that when I created the url is I would then have to remove it again when I check the gtg's